public/app/features/dashboard-scene/sharing/ShareButton/share-snapshot/CreateSnapshot.tsx

```diff
@@ -0,0 +1,98 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { SceneComponentProps } from '@grafana/scenes';
+import {
+  Alert,
+  Divider,
+  Field,
+  RadioButtonGroup,
+  Spinner,
+  Stack,
+  TextLink,
+  Text,
+  Button,
+  useStyles2,
+} from '@grafana/ui';
+import { Input } from '@grafana/ui/src/components/Input/Input';
+import { t } from '@grafana/ui/src/utils/i18n';
+import { Trans } from 'app/core/internationalization';
+
+import { getExpireOptions } from '../../ShareSnapshotTab';
+
+import { ShareSnapshot } from './ShareSnapshot';
+
+const SNAPSHOT_URL = 'https://grafana.com/docs/grafana/latest/dashboards/share-dashboards-panels/#publish-a-snapshot';
+
+interface Props extends SceneComponentProps<ShareSnapshot> {
+  onCreateClick: (isExternal?: boolean) => void;
+  isLoading: boolean;
+}
+export function CreateSnapshot({ model, onCreateClick, isLoading }: Props) {
+  const { snapshotName, selectedExpireOption, dashboardRef, snapshotSharingOptions } = model.useState();
+  const styles = useStyles2(getStyles);
```

Comment:
This feels a bit wrong to me, I would not use SceneComponentProps for this component. IMO this should be a dummy, stateless component and the `model.useState` should be lifted to `ShareSnapshotRenderer`. `onClickCancel` could be provided as a property to avoid dependency on `dashboardRef`.

Reply:
so, you mean passing the rest of the values as props instead of passing the whole model, right? 
`snapshotName`, `selectedExpireOption` and so on

Reply:
I think so, then you can test this component independently from scenes., also re-use it if needed.